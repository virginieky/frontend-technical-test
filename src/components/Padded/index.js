import PropTypes from 'prop-types';
import styled from 'styled-components';

const Padded = styled.div`
  position: relative;
  padding-top: ${({ top, small }) => top && (!small ? '10px' : '5px')};
  padding-right: ${({ right, small }) => right && (!small ? '10px' : '5px')};
  padding-bottom: ${({ bottom, small }) => bottom && (!small ? '10px' : '5px')};
  padding-left: ${({ left, small }) => left && (!small ? '10px' : '5px')};
`;

Padded.defaultProps = {
  bottom: false,
  left: false,
  right: false,
  small: false,
  top: false,
};

Padded.propTypes = {
  bottom: PropTypes.bool,
  left: PropTypes.bool,
  right: PropTypes.bool,
  small: PropTypes.bool,
  top: PropTypes.bool,
};

export default Padded;
