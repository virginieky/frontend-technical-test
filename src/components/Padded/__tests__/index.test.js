import React from 'react';
import { render } from '@testing-library/react';
import Padded from '../index';

describe('components | Padded', () => {
  it('should match the snapshot', () => {
    const { asFragment } = render(<Padded />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should match the snapshot with attributes', () => {
    const { asFragment } = render(<Padded bottom left right small top />);

    expect(asFragment()).toMatchSnapshot();
  });
});
