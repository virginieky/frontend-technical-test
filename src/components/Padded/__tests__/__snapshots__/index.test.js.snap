// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components | Padded should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="Padded-sc-q6j7xl-0 llxRKk"
  />
</DocumentFragment>
`;

exports[`components | Padded should match the snapshot with attributes 1`] = `
<DocumentFragment>
  <div
    class="Padded-sc-q6j7xl-0 cUZIng"
  />
</DocumentFragment>
`;
